Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -207,14 +209,67 @@ namespace PhysX
         *m_shapeConfig = Utils::CreateHeightfieldShapeConfiguration(GetEntityId());
     }
 
-    void EditorHeightfieldColliderComponent::RefreshHeightfield()
+    void EditorHeightfieldColliderComponent::RefreshHeightfield(const AZ::Aabb& dirtyRegion,
+        [[maybe_unused]] const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask* changeMask)
     {
-        ClearHeightfield();
-        InitHeightfieldShapeConfiguration();
+        AZ::Aabb heightfieldAabb = GetColliderShapeAabb();
+        AZ::Aabb requestRegion = dirtyRegion;
+
+        if (!requestRegion.IsValid())
+        {
+            requestRegion = heightfieldAabb;
+        }
+
+        // Early out if the updated region is outside of the heightfield Aabb
+        if (heightfieldAabb.Disjoint(requestRegion))
+        {
+            return;
+        }
+
+        // Clamp requested region to the entire heightfield AABB
+        requestRegion = heightfieldAabb.GetClamped(requestRegion);
 
-        if (!m_shapeConfig->GetSamples().empty())
+        // if dirty region invalid, recreate the entire heightfield, otherwise request samples
+        bool shouldRecreateHeightfield = m_shapeConfig == nullptr;
+
+        // Check if dirtyRegion covers the entire terrain
+        shouldRecreateHeightfield = shouldRecreateHeightfield || (requestRegion == heightfieldAabb);
+
+        // Check if base configuration parameters have changed
+        if (!shouldRecreateHeightfield)
         {
-            InitStaticRigidBody();
+            Physics::HeightfieldShapeConfiguration baseConfiguration = Utils::CreateBaseHeightfieldShapeConfiguration(GetEntityId());
+            shouldRecreateHeightfield = shouldRecreateHeightfield || (baseConfiguration.GetNumRows() != m_shapeConfig->GetNumRows());
+            shouldRecreateHeightfield = shouldRecreateHeightfield || (baseConfiguration.GetNumColumns() != m_shapeConfig->GetNumColumns());
+            shouldRecreateHeightfield = shouldRecreateHeightfield || (baseConfiguration.GetMinHeightBounds() != m_shapeConfig->GetMinHeightBounds());
+            shouldRecreateHeightfield = shouldRecreateHeightfield || (baseConfiguration.GetMaxHeightBounds() != m_shapeConfig->GetMaxHeightBounds());
```

Comment:
This could be 1 statement:

````
shouldRecreateHeightfield =
    (baseConfiguration.GetNumRows() != m_shapeConfig->GetNumRows()) ||
    (baseConfiguration.GetNumColumns() != m_shapeConfig->GetNumColumns()) ||
    (baseConfiguration.GetMinHeightBounds() != m_shapeConfig->GetMinHeightBounds()) ||
    (baseConfiguration.GetMaxHeightBounds() != m_shapeConfig->GetMaxHeightBounds());
````